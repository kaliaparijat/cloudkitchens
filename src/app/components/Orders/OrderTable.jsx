import React from 'react';
import capitalize from 'lodash.capitalize';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
  },
  paper: {
    marginTop: theme.spacing(3),
    width: '100%',
    overflowX: 'auto',
    marginBottom: theme.spacing(2),
  },
  table: {
    minWidth: 650,
  },
  button: {
    border: 'none',
  }
}));

export const SimpleOrderTable = (props) => {
  const classes = useStyles();
  return (
    <Table className={classes.table}>
      <TableHead>
        <TableRow>
          <TableCell align="left">Status</TableCell>
          <TableCell align="left">Food item</TableCell>
          <TableCell align="left">Order location</TableCell>
          <TableCell align="left">Actions</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {props.orders.map((order) => (
          <TableRow key={order.id} data-testid={order.id} >
            <TableCell align="left">
              { props.editingOrderId !== order.id ?
                capitalize(order.event_name.toLowerCase().replace('_', ' ')) :
                (
                  <Select
                    value={order.event_name}
                    onChange={props.handleOrderUpdate}
                    aria-label={`select-menu-${order.id}`}
                  >
                    <MenuItem value="CREATED">Created</MenuItem>
                    <MenuItem value="COOKED">Cooked</MenuItem>
                    <MenuItem value="DRIVER_RECEIVED">Driver received</MenuItem>
                    <MenuItem value="DELIVERED">Delivered</MenuItem>
                    <MenuItem value="CANCELLED">Cancelled</MenuItem>
                  </Select>
                )
              }
            </TableCell>
            <TableCell align="left">{order.name}</TableCell>
            <TableCell align="left">{order.destination}</TableCell>
            <TableCell align="left">
              <button
                onClick={() => props.handleEdit(order.id)}
                aria-label={`action-button-${order.id}`}
              >
                {props.editingOrderId === order.id? 'Done' : 'Edit'}
              </button>
            </TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  )
};

