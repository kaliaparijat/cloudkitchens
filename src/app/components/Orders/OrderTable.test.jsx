import React from 'react';
import ReactDOM from 'react-dom';
import { render, fireEvent } from '@testing-library/react';

import { SimpleOrderTable }  from "./OrderTable";

const sortedOrders = [
    {
        "destination": "801 Toyopa Dr, Pacific Palisades, CA 90272",
        "event_name": "CREATED",
        "id": "4b76edxf",
        "name": "Cheese pizza",
        "sent_at_second": 4
    },
    {
        "destination": "801 Toyopa Dr, Pacific Palisades, CA 90272",
        "event_name": "CANCELLED",
        "id": "4b76edzf",
        "name": "Cheese pizza",
        "sent_at_second": 10
    },
    {
        "destination": "801 Toyopa Dr, Pacific Palisades, CA 90272",
        "event_name": "DELIVERED",
        "id": "4b76edbf",
        "name": "Cheese pizza",
        "sent_at_second": 17
    },
];


describe('SimpleOrderTable with no editingOrderId prop', () => {

    const handleOrderEdit = jest.fn();

    let renderedOrderTable;

    beforeEach(() => {
        renderedOrderTable = render(<SimpleOrderTable orders={sortedOrders} handleEdit={handleOrderEdit}/>);
    });

    it('should invoke handleEdit correctly with the right order ID parameter when the user clicks on an Edit button', () => {
        const { getByTestId, asFragment } = renderedOrderTable;
        const editButton = getByTestId('action-button-4b76edxf');
        fireEvent.click(editButton);
        expect(handleOrderEdit).toHaveBeenCalled();
        expect(handleOrderEdit).toHaveBeenCalledWith('4b76edxf');
    });

    describe ('SimpleOrderTable with an editingOrderId prop', () => {
        const handleOrderStatusUpdate = jest.fn();
        let selectMenu;

        beforeEach(() => {
           renderedOrderTable = render(
             <SimpleOrderTable
             orders={sortedOrders}
             handleEdit={handleOrderEdit}
             handleUpdate={handleOrderStatusUpdate}
             editingOrderId='4b76edxf'/>);
           const { getByLabelText } = renderedOrderTable;
           selectMenu = getByLabelText('select-menu-4b76edxf');
        });

        it('should render a select menu under the status column for the order that has the orderID same as editingOrderId', () => {
            expect(selectMenu).toBeTruthy();
        });
    });

});
