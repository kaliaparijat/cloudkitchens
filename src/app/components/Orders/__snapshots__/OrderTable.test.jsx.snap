]// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleOrderTable it receives an array of orders as a prop and renders them as a table 1`] = `
<DocumentFragment>
  <table
    class="MuiTable-root makeStyles-table-3"
  >
    <thead
      class="MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-head"
          scope="col"
        >
           Order For
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-head"
          scope="col"
        >
          Current status
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-head"
          scope="col"
        >
          Order name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-head"
          scope="col"
        >
          Change order status
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root"
        data-testid="4b76edxf"
      >
        <td
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-body"
        >
          Cheese pizza
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-body"
        >
          801 Toyopa Dr, Pacific Palisades, CA 90272
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-body"
        >
          Created
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
        >
          <button
            class="makeStyles-button-4"
            value="4b76edxf"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root"
        data-testid="4b76edzf"
      >
        <td
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-body"
        >
          Cheese pizza
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-body"
        >
          801 Toyopa Dr, Pacific Palisades, CA 90272
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-body"
        >
          Cancelled
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
        >
          <button
            class="makeStyles-button-4"
            value="4b76edzf"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root"
        data-testid="4b76edbf"
      >
        <td
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-body"
        >
          Cheese pizza
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-body"
        >
          801 Toyopa Dr, Pacific Palisades, CA 90272
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-body"
        >
          Delivered
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-body"
        >
          <button
            class="makeStyles-button-4"
            value="4b76edbf"
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
